Services/UICore/classes/Setup/class.ilCtrlStructureSetupAgent.php

```diff
@@ -0,0 +1,65 @@
+<?php declare(strict_types=1);
+
+use ILIAS\Setup;
+use ILIAS\Refinery\Transformation;
+
+class ilCtrlStructureSetupAgent implements Setup\Agent
+{
+	/**
+	 * @var ilCtrlStructureReader
+	 */
+	protected $ctrl_reader;
+
+	public function __construct(\ilCtrlStructureReader $ctrl_reader)
+	{
+		$this->ctrl_reader = $ctrl_reader;
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function hasConfig() : bool
+	{
+		return false;
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function getConfigInput(Setup\Config $config = null): ILIAS\UI\Component\Input\Field\Input
+	{
+		throw new \LogicException(self::class." has no Config.");
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function getArrayToConfigTransformation(): Transformation
+	{
+		throw new \LogicException(self::class." has no Config.");
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function getInstallObjective(Setup\Config $config = null): Setup\Objective
+	{
+		return new \ilCtrlStructureStoredObjective($this->ctrl_reader);
+	}
+
+	/**
+	 * @inheritdoc
+	 */
+	public function getUpdateObjective(Setup\Config $config = null): Setup\Objective
+	{
+		return new Setup\NullObjective();
```

Comment:
```suggestion
		return new \ilCtrlStructureStoredObjective($this->ctrl_reader);
```

I see no harm in reloading the control structure on every update. This will get rid of the requirement that devs need to explicitly do that.